cogs/vip.py

```diff
@@ -18,6 +18,41 @@
 class VIP(Cog):
     cog_command_error = cog_error_handler
 
+    #check if vip is still valid
+	#this is probably going to need to be threaded?
+    @Cog.listener()
+    async def on_message(self, message):
+
+        #vip role
+        vip_role = discord.utils.get(message.guild.roles, name="VIP")
+
+        #stop bot from replying to itself :)
+        if message.author.bot:
+            return
+		
+        #don't bother checking if they are still a vip if they don't have the role in the first place...
+        if vip_role in message.author.roles:
+
+            #connect to xf database
+            database = databases.Database(global_config.databases.tf2maps_site)
+            await database.connect()
+
+            #check if discord id is linked
+            query = "SELECT user_id FROM xf_user_field_value WHERE field_id = :field_id AND field_value = :field_value"
+            values = {"field_id": "discord_user_id", "field_value": message.author.id}
+            result = await database.fetch_one(query=query, values=values)
+
+            #check if they are vip
+            query = "SELECT secondary_group_ids FROM xf_user WHERE user_id = :user_id AND find_in_set(:vip_gid, secondary_group_ids)"
+            values = {"user_id": result[0], "vip_gid": 19}
```

Comment:
if the previous value of result doesnt return any data; this will throw an unhandled exception

Reply:
FIxed. 
![image](https://user-images.githubusercontent.com/44575567/141389788-26654c31-431c-4847-9cef-e15cdd1ffe9e.png)